lib/diagnostic.js

```diff
@@ -8,30 +8,50 @@
 // and returns the minimum from that set
 // assume that all the argument are numbers
 const min = function min() {
-
+  //returns the minimum amount of arguments? or the smallest interger?
 };
 
 // Question 2
 // Write a function that takes zero or more arguments
 // and returns the sum of that set
 // assume that all the argument are numbers
 const sum = function sum() {
+  let total = 0;
+  for (let i = 0; i < arguments.length; i++) {
+    total += arguments[i];
+  }
 
+  return total;
 };
 
+sum([1, 2, 3, 4]);
+
 // Question 3
 // Write a function that returns an array of the specified length
 // with all the elements set to defaultValue
-const newArray = function newArray(/* length, defaultValue */) {
+const newArray = function newArray(length, defaultValue) {
+  let array = [];
+  for (let i = 0; i < length; i++) {
+    array.push[i](defaultValue);
```

Comment:
```push``` is an array method that adds an element onto the end of the array.  This should be ```array.push(defaultValue);```.